sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Conditions/ComparisonCondition.cs

```diff
@@ -37,19 +35,44 @@ public ComparisonCondition(string field, QueryComparisonOperator comparisonOpera
 
         public override string Stringify()
         {
+            // TODO -- generic type on this class to handle different inputs
             var conditionString = new StringBuilder();
             conditionString.Append($"{Field} {QueryConstants.ComparisonOperators[Operator]} ");
 
             // check to see if the input is numeric value -- if not, we need single quotes around Value
-            bool isNumeric = int.TryParse(Value, out int numericValue);
+            bool isNumeric;
+            bool quoteSurrounded = (Value[0] == '\'' && Value[Value.Length - 1] == '\'');
 
-            if (!isNumeric)
+            // check to see if the user has entered single quotes around the Value
+            if (quoteSurrounded)
+            {
+                isNumeric = int.TryParse(Value.Substring(1, Value.Length - 2), out _);
+            }
+            else
             {
-                // surround with single quotes
-                Value = "'" + Value + "'";
+                isNumeric = int.TryParse(Value, out _);
             }
 
-            conditionString.Append(Value);
+            if (!isNumeric)
+            {
+                if (!quoteSurrounded)
+                {
+                    // surround with single quotes
+                    Value = "'" + Value + "'";
+                }
+
+                conditionString.Append(Value);
+            }
+            // if not numeric but surrounded by quotes, it must be a number surrounded by quotes (which must be removed)
+            else if (quoteSurrounded)
+            {
+                conditionString.Append(Value.Substring(1, Value.Length - 2));
```

Comment:
Here's that code snippet again. DRY violation.

I can't tell what you are trying to do. This means code comment, or private method with a good name. Given DRY violation, I'd go for private method.